pyteal/ast/box.py

```diff
@@ -0,0 +1,226 @@
+from typing import TYPE_CHECKING
+from pyteal.ast.maybe import MaybeValue
+from pyteal.errors import TealInputError
+
+from pyteal.types import TealType, require_type
+from pyteal.ir import TealOp, Op, TealBlock
+from pyteal.ast.expr import Expr
+
+if TYPE_CHECKING:
+    from pyteal.compiler import CompileOptions
+
+
+class BoxCreate(Expr):
+    """Create a box with a given name and size."""
+
+    def __init__(self, name: Expr, size: Expr) -> None:
+        """
+        Args:
+            name: The key used to reference this box. Must evaluate to a bytes.
+            size: The number of bytes to reserve for this box. Must evaluate to a uint64.
+        """
+
+        super().__init__()
+        require_type(name, TealType.bytes)
+        require_type(size, TealType.uint64)
+        self.name = name
+        self.size = size
+
+    def __teal__(self, options: "CompileOptions"):
+        if options.version < Op.box_create.min_version:
```

Comment:
@ahangsu Few minor thoughts here:
* Via https://github.com/algorand/pyteal/issues/392, we'll soon consolidate TEAL and AVM versioning.  While I'm not asking to pick up the ticket now, I prefer to avoid adding more _TEAL version_ references.
  * As the ticket suggests, depending on the context, _program version_ or _AVM version_ might be better fits. 
* While looking here, it occurred to me a common, helper function exists:  `errors.py::verifyTealVersion`.  Can it be reused?
  * Optional asides - Feel welcomed to disregard:
    * It seems to me the usage of `msg` by callers often duplicates the internal error message.  If you're feeling ambitious, we can discuss normalization options (I see several).
    * When invoking `verifyTealVersion`, I'd prefer keyword arguments for `minVersion` and `version`.  Since the types are the same, I think it's easy to mistakenly invert argument sequencing.  
      * Obviously, a test _should_ exist.  I'm suggesting an optional, secondary safety measure.

Reply:
good point on reusing `verifyTealVersion`, I swapped all implementation to follow such pattern.

On the argument issues for `verifyTealVersion`, if we use keyword arguments for `minVersion` and `version`, then we would have to change a few internal code, and I would like that happening in another separate PR into master.

I am unaware of the consolidating of Teal and AVM version, if AVM version is preferred, then I will try to reflect it in error msg and variable naming.

Reply:
@ahangsu I'm good to resolve.  Thanks for the `verifyTealVersion` change + AVM version consolidation consideration.  Leaving a clarification for informational purposes.

> if we use keyword arguments for minVersion and version, then we would have to change a few internal code

Sorry - I think my vocabulary might be inaccurate.  I'm _not_ considering a change to `verifyTealVersion` implementation.  I'm trying to suggest invoking with named parameters like:

```
verifyTealVersion(
            minVersion = Op.box_del.min_version,
            version = options.version,
            f"{Op.box_del} unavailable for low version.",
)
```